app/src/main/java/com/google/samples/apps/nowinandroid/MainActivity.kt

```diff
@@ -95,9 +99,28 @@ class MainActivity : ComponentActivity() {
         WindowCompat.setDecorFitsSystemWindows(window, false)
 
         setContent {
-            val systemUiController = rememberSystemUiController()
+            val uiModeManager = getSystemService(Context.UI_MODE_SERVICE) as UiModeManager
             val darkTheme = shouldUseDarkTheme(uiState)
 
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
```

Comment:
Since this is a side-effect of composition, we should wrap this into some sort of `SideEffect` or `DisposableEffect` block.

Reply:
@tunjid do you think there is a better place to put this logic? It doesn't have to be done as a side-effect of composition, I think it could be some sort of global listener that is observing the user preferences.

Reply:
I think so too. Seems `NiaAppState` is a better candidate for this. It can be passed a function to invoke when it detects a theme change has occured.

Reply:
Moved to `NiaAppState`